In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline 

In [2]:
uniq_id_cust    = pd.read_csv ('/mnt/HC_Volume_18315164/home-jupyter/jupyter-a-lesihina/first_project/olist_customers_dataset.csv')
orders          = pd.read_csv ('/mnt/HC_Volume_18315164/home-jupyter/jupyter-a-lesihina/first_project/olist_orders_dataset.csv')
items           = pd.read_csv ('/mnt/HC_Volume_18315164/home-jupyter/jupyter-a-lesihina/first_project/olist_order_items_dataset.csv')

FileNotFoundError: [Errno 2] No such file or directory: '/mnt/HC_Volume_18315164/home-jupyter/jupyter-a-lesihina/first_project/olist_customers_dataset.csv'

#### Задание 1. Сколько пользователей совершили покупку 1 раз

In [4]:
#объединим таблицу с id пользователей и таблицу с заказами
join_id_orders = uniq_id_cust.merge(orders, how = 'outer', on ='customer_id')

In [5]:
join_id_orders.head ()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  \
0                     14409                 franca             SP   
1                      9790  sao bernardo do campo             SP   
2                      1151              sao paulo             SP   
3                      8775        mogi das cruzes             SP   
4                     13056               campinas             SP   

                           order_id order_status order_purchase_timestamp  \
0  00e7ee1b050b8499577073aeb2a297a1    delivered      2017-05-16 15:05:35   
1  29150127e6685892b6eab3eec79f59c7    delivered      2018-01-12 20:48:24   
2  b2059ed67ce144a36e2aa97d2c9e9ad2    delivered      2018-05-19 16:07:45   
3  951670f92359f4fe4a63112aa7306eba    delivered      2018-03-13 16:06:38   
4  6b7d50bd145f6fc7f33cebabd7e49d0f    delivered      2018-07-29 09:51:30   

     order_approved_at order_delivered_carrier_date  \
0  2017-05-16 15:22:12          2017-05-23 10:47:57   
1  2018-01-12 20:58:32          2018-01-15 17:14:59   
2  2018-05-20 16:19:10          2018-06-11 14:31:00   
3  2018-03-13 17:29:19          2018-03-27 23:22:42   
4  2018-07-29 10:10:09          2018-07-30 15:16:00   

  order_delivered_customer_date order_estimated_delivery_date  
0           2017-05-25 10:35:35           2017-06-05 00:00:00  
1           2018-01-29 12:41:19           2018-02-06 00:00:00  
2           2018-06-14 17:58:51           2018-06-13 00:00:00  
3           2018-03-28 16:04:25           2018-04-10 00:00:00  
4           2018-08-09 20:55:48           2018-08-15 00:00:00

In [6]:
# посмотрим сколько заказов сделал каждый клиент
all_orders = join_id_orders\
    .groupby ('customer_unique_id', as_index =False)\
    .agg ({'order_id': 'count'})

In [13]:
all_orders.max ()

customer_unique_id    ffffd2657e2aad2907e67c3e9daecbeb
order_id                                            17
dtype: object

In [14]:
all_orders

customer_unique_id  order_id
0      0000366f3b9a7992bf8c76cfdf3221e2         1
1      0000b849f77a49e4a4ce2b2a4ca5be3f         1
2      0000f46a3911fa3c0805444483337064         1
3      0000f6ccb0745a6a4b88665a16c9f078         1
4      0004aac84e0df4da2b147fca70cf8255         1
...                                 ...       ...
96091  fffcf5a5ff07b0908bd4e2dbc735a684         1
96092  fffea47cd6d3cc0a88bd621562a9d061         1
96093  ffff371b4d645b6ecea244b27531430a         1
96094  ffff5962728ec6157033ef9805bacc48         1
96095  ffffd2657e2aad2907e67c3e9daecbeb         1

[96096 rows x 2 columns]

In [8]:
'''
Посмотрим сколько заказов сделал каждый клиент 
(кроме неоплаченных, не созданных, не доступных и отмененных: created, invoiced, unavailable, canceled)
'''
select_orders = join_id_orders\
    .query ('order_status != ["created", "invoiced", "unavailable", "canceled"]')\
    .groupby ('customer_unique_id', as_index =False)\
    .agg ({'order_id': 'count'})

In [9]:
select_orders

customer_unique_id  order_id
0      0000366f3b9a7992bf8c76cfdf3221e2         1
1      0000b849f77a49e4a4ce2b2a4ca5be3f         1
2      0000f46a3911fa3c0805444483337064         1
3      0000f6ccb0745a6a4b88665a16c9f078         1
4      0004aac84e0df4da2b147fca70cf8255         1
...                                 ...       ...
94683  fffcf5a5ff07b0908bd4e2dbc735a684         1
94684  fffea47cd6d3cc0a88bd621562a9d061         1
94685  ffff371b4d645b6ecea244b27531430a         1
94686  ffff5962728ec6157033ef9805bacc48         1
94687  ffffd2657e2aad2907e67c3e9daecbeb         1

[94688 rows x 2 columns]

In [15]:
select_orders.max()

customer_unique_id    ffffd2657e2aad2907e67c3e9daecbeb
order_id                                            16
dtype: object

In [10]:
# Посчитаем количество клиентов,которые сделали только 1 заказ (любой по статусу)
all_orders.query ('order_id == 1').count()

customer_unique_id    93099
order_id              93099
dtype: int64

In [11]:
# Посчитаем количество клиентов,которые сделали только 1 заказ (только успешные)
select_orders.query ('order_id == 1').count()

customer_unique_id    91816
order_id              91816
dtype: int64

Ответ 1: Всего 93099 пользователей,которые совершили (любую) покупку один раз. 

Если учитывать только успешные (оплаченные) заказы, то кол-во пользователей 91816.